include/engine/api/route_parameters.hpp

```diff
@@ -202,8 +202,8 @@ inline RouteParameters::AnnotationsType operator|(RouteParameters::AnnotationsTy
         static_cast<std::underlying_type_t<RouteParameters::AnnotationsType>>(rhs));
 }
 
-inline RouteParameters::AnnotationsType operator|=(RouteParameters::AnnotationsType lhs,
-                                                   RouteParameters::AnnotationsType rhs)
+inline RouteParameters::AnnotationsType &operator|=(RouteParameters::AnnotationsType &lhs,
+                                                    RouteParameters::AnnotationsType rhs)
```

Comment:
Might be worth deleting this function instead. I assume given the bug that it's not in use.

Reply:
I can commit a change to remove it instead if it's preferred, but I think that given the use of the `|` operator, it might be more helpful to keep around fixed, since it can help when calculating the AnnotationsType (ie. able to do `lhs |= rhs` rather than `lhs = lhs | rhs`).

Reply:
Sounds good, thanks for spotting :+1: